### PR TITLE
POL-229-AWS Schedule Instance

### PR DESCRIPTION
**Pull request by KVivek03**
### Description

POL-229-AWS Schedule Instance
This Policy Template allows user to schedule start and stop times for your instance, along with the option to terminate instance, update and delete schedule.     

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] New functionality has been documented in CHANGELOG.MD

https://governance.rightscale.com/org/1105/projects/60073/applied-policies/5ec4ecdbc72da20001e4ea09
image

### CODE REVIEW

File: cost/aws/schedule_instance/aws_schedule_instance.pt [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+    end
+  end
+end
+
+###############################################################################
+# Cloud Workflow
+###############################################################################
+
+define schedule_instance($data, $action) return $all_response do
+  $$debug=true
```
Comment (cdwilhelm) [inline cost/aws/schedule_instance/aws_schedule_instance.pt:228]: you can remove $$debug

File: cost/aws/schedule_instance/aws_schedule_instance.pt [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+###############################################################################
+
+escalation "esc_email" do
+  automatic true
+  label "Sends Email"
+  description "Sends Incident Email"
+  email $param_email
+end
+
+escalation "esc_action" do
```
[the remark this answers is not in the document]
Comment (KVivek03) [reply cost/aws/schedule_instance/aws_schedule_instance.pt]: made the change

File: cost/aws/schedule_instance/aws_schedule_instance.pt [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+      field "region" do
+        label "Region"
+      end
+      field "id" do
+        label "Instance ID"
+      end
+      field "state" do
+        label "Instance State"
+      end
+      field "schedule" do
```
[the remark this answers is not in the document]
Comment (KVivek03) [reply cost/aws/schedule_instance/aws_schedule_instance.pt:243]: added Tags column

File: cost/aws/schedule_instance/aws_schedule_instance.pt [hunk 4]
```diff
@@ -0,0 +1,444 @@
...
+  $body = $response['body']
+  call sys_log('window active $body', to_s($body))
+
+  $window_active = to_b($body['event_active'])
+  $next_start   = $body['next_start']
+  $next_stop    = $body['next_stop']
+end
+
+define sys_log($subject, $detail) do
+  if $$debug
```
[the remark this answers is not in the document]
Comment (KVivek03) [reply cost/aws/schedule_instance/aws_schedule_instance.pt]: removed $$debug

File: cost/aws/schedule_instance/aws_schedule_instance.pt (hunk 1 above)
Comment (KVivek03) [reply cost/aws/schedule_instance/aws_schedule_instance.pt]: removed $$debug

File: cost/aws/schedule_instance/README.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+## What it does
+
+This Policy Template allows you to schedule start and stop times for your instance, along with the option to terminate instance, update and delete schedule.         
+
+## How to Use
+
+This policy relies on a tag with format 'schedule' to stop and start instances based on a schedule. The tag value defines the schedule with a start hour, stop hour and days of the week. The start and stop hour are in 24 hour format, and the days of the week are two character abbreviation for example: MO, TU, WE. See full example below.. Use a Timezone TZ value to indicate a timezone to stop/start the instance(s) 
+
+## schedule Tag Example
```
Comment (cdwilhelm) [inline cost/aws/schedule_instance/README.md:11]: I think schedule should be Schedule

File: cost/aws/schedule_instance/README.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+- *Email addresses* - A list of email addresses to notify  
+- *Tags Key=Value* - List of tags that an Instance can have to exclude it from the list. 
+
+## Policy Actions 
+
+The following policy actions are taken on any resources found to be out of compliance.
+
+- Send an email report  
+- stop  - stop a selected instance
+- start - start a selected instance
```
Comment (cdwilhelm) [inline cost/aws/schedule_instance/README.md:36]: I think these two actions should be condensed to one schedule action.

File: cost/aws/schedule_instance/README.md (hunk 5 above)
Comment (KVivek03) [reply cost/aws/schedule_instance/README.md]: made the change

File: cost/aws/schedule_instance/README.md (hunk 6 above)
Comment (KVivek03) [reply cost/aws/schedule_instance/README.md]: made the change

**Comment by cdwilhelm** [comment]
@KVivek03  this PR needs to be updated with the automatic actions parameter

**Comment by KVivek03** [comment]
Hi @cdwilhelm , I have updated the PR with the automatic parameter.

**Pull request merged**